build/accelerate_launch.py

```diff
@@ -47,6 +48,40 @@
 ERROR_LOG = "/dev/termination-log"
 
 
+def copytree(source, destination, symlinks=False, ignore=None):
+    if not os.path.exists(destination):
+        os.makedirs(destination)
+        shutil.copystat(source, destination)
+    file_list = os.listdir(source)
+    # If we want to ignore any files
+    if ignore:
+        exclude = ignore(source, file_list)
+        file_list = [a_file for a_file in file_list if a_file not in exclude]
+    # Have a list of directory objects, now iterate over them.
+    for item in file_list:
+        source_file = os.path.join(source, item)
+        destination_file = os.path.join(destination, item)
+        if symlinks and os.path.islink(source_file):
+            if os.path.lexists(destination_file):
+                os.remove(destination_file)
+            os.symlink(os.readlink(source_file), destination_file)
+            # Improvement over shutil.copytree: BAD CHMOD IS NOT FATAL.
+            try:
+                # get the permissions on the source file
```

Comment:
The issue we were seeing is with the directory permissions, not the file permissions. I don't think the file permissions are an issue at all and copytree already used `copy2()` to copy the files over. 

If this is symlinking the file, does that not already copy the permissions of the file over?

Reply:
Also I don't think you need to worry about symlinks because `shutil.copy()` and `shutil.copy2()` and `shutil.copyfile()` all allow the option to set `follow_symlinks=True` and covers this for you